### ISSUE TITLE
underflow encountered when using RescaleIntensity

### ISSUE THREAD

**Issue by GabriellaKamlish** [fepegar/torchio#502]
**🐛Bug**

**To reproduce**

Download this file, and attempt to rescale between -1, and 1. 
[output.nii.gz](https://github.com/fepegar/torchio/files/6304166/output.nii.gz)

```python
import torchio as tio
import nibabel as nib

# PATH to output.nii.gz
Resection_name = 'output.nii.gz'
Resection_img = tio.ScalarImage(Resection_name)

Rescale = tio.RescaleIntensity( (-1,1) )

Resection_img = Rescale(Resection_img)
```

**Expected behavior**

Expected to Resection_img.data to be a torch tensor which has minimum -1 and maximum of 1

**Actual behavior**

```python-traceback
Traceback (most recent call last):
  File "test.py", line 19, in <module>
    Resection_img = Rescale(Resection_img)
  File "/Users/gabriellakamlish/anaconda3/envs/my_Env/lib/python3.8/site-packages/torchio/transforms/transform.py", line 121, in __call__
    transformed = self.apply_transform(subject)
  File "/Users/gabriellakamlish/anaconda3/envs/my_Env/lib/python3.8/site-packages/torchio/transforms/preprocessing/intensity/normalization_transform.py", line 53, in apply_transform
    self.apply_normalization(subject, image_name, mask)
  File "/Users/gabriellakamlish/anaconda3/envs/my_Env/lib/python3.8/site-packages/torchio/transforms/preprocessing/intensity/rescale.py", line 54, in apply_normalization
    image.set_data(self.rescale(image.data, mask, image_name))
  File "/Users/gabriellakamlish/anaconda3/envs/my_Env/lib/python3.8/site-packages/torchio/transforms/preprocessing/intensity/rescale.py", line 77, in rescale
    array /= array_max  # [0, 1]
FloatingPointError: underflow encountered in true_divide
```

**System info**

Output of `python <(curl -s https://raw.githubusercontent.com/fepegar/torchio/master/print_system.py)`:

```
Platform: macOS-10.13.6-x86_64-i386-64bit
TorchIO:  0.18.29
PyTorch:  1.6.0
NumPy:    1.19.4
Python:   3.8.5 (default, Sep  4 2020, 02:22:02) 
[Clang 10.0.0 ]
```

**Comment by fepegar** [comment]
Hi, @GabriellaKamlish. Thanks for reporting. I have been able to reproduce this.

This bug it's caused by
https://github.com/fepegar/torchio/blob/f9b9d85593de52ecd0afea4208cef2d04975d662/torchio/transforms/transform.py#L120-L121

According to NumPy docs ([How numpy handles numerical exceptions](https://jiffyclub.github.io/numpy/user/misc.html#how-numpy-handles-numerical-exceptions), the default action for underflows is to ignore them, but we are telling NumPy to raise an exception instead. A solution would be to add a kwarg `under='ignore'` to that context manager call.

Would you like to contribute with a pull request?

**Comment by fepegar** [comment]
By the way, it looks like you used nearest neighbor interpolation for that image. I suggest you use `'linear'` or, if it's precomputed (offline), `'bspline'`.

**Comment by romainVala** [comment]
it looks like  #327 
we try to understand what happening, but could not find what cause the underflow
 (not NaN in the data nor strange range ...)

I see that you fix by ignoring it,  #503 
but any idea where it come from ?

**Comment by fepegar** [comment]
@romainVala you're right! It is the same error. Apparently, an underflow happens when a value is too small to be represented by the computer. I suppose the implemented workaround is to use the actual smallest number that can be represented. Anyway, it's very low-level and doesn't seem to be a problem for us.

---

@GabriellaKamlish, this will be solved in `v0.18.33`. Please upgrade and try again.

---

@all-contributors please add @GabriellaKamlish for bug
